core/src/toga/app.py

```diff
@@ -541,9 +541,18 @@ def _startup(self) -> None:
         self.startup()
         self._verify_startup()
 
-        # Manifest the initial state of the menus.
+        # Manifest the initial state of the menus. This will cascade down to all
+        # open windows if the platform has window-based menus.
         self._impl.create_menus()
 
+        # Manifest the initial state of toolbars (on the windows that have
+        # them), then install a change listener so that any future changes to
+        # the toolbar cause a change in toolbar items.
```

Comment:
Having the code here is an optimisation. Items are always added to the menu and toolbar after the CommandSet for those objects have been created. If the `on_change` handler is in place when the toolbar is created, this results in creating an empty toolbar, then *recreating* the toolbar every time you add a new command. It works, and the extra work isn't the end of the world, but it does slow down app startup (marginally).

A previous iteration tackled this by added a "suspend notifications" capability to CommandSet. However, it turns out the only time you actually need to suspect notifications is during app startup, so it's simpler to just not install the on_change handler until you've completed startup. 

The `hasattr` is needed because I wanted to avoid adding a `create_toolbar()` method on toga.Window, because that gives the impression that it *could* have a toolbar (similarly for calling window-level `create_menus()`). 

Putting the toolbar creation code between the menu creation code and the menu's on_change handler is definitely an oversight, though.